streaming/src/main/scala/org/apache/spark/streaming/dstream/ConstantInputDStream.scala

```diff
@@ -27,6 +28,9 @@ import scala.reflect.ClassTag
 class ConstantInputDStream[T: ClassTag](ssc_ : StreamingContext, rdd: RDD[T])
   extends InputDStream[T](ssc_) {
 
+  require(rdd != null,
```

Comment:
(This could have been resolved with https://github.com/apache/spark/pull/8881 for more consistency rather than a different patch)

Is this inconsistent with SPARK-10772 and the other changes, which cause `SparkException`?